From 297862622c345fecbf4169b7e7fc8c89beb983f8 Mon Sep 17 00:00:00 2001
From: Christophe Kerello <christophe.kerello@foss.st.com>
Date: Tue, 1 Jun 2021 09:38:47 +0200
Subject: [PATCH 121/954] feat(stm32mp25-fdts): add OSPI support

This patch adds OSPI1 and OSPI2 support on STM32MP257.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 6f1a384f40e32e621a460fda51acf7714469602d feat(stm32mp25-fdts): add OSPI support]
Change-Id: I2a072f220cb416e4ffcdce1651e22d7fe79ebb4f
Signed-off-by: Christophe Kerello <christophe.kerello@foss.st.com>
---
 fdts/stm32mp25-bl2.dtsi |  4 ++++
 fdts/stm32mp251.dtsi    | 22 ++++++++++++++++++++++
 2 files changed, 26 insertions(+)

diff --git a/fdts/stm32mp25-bl2.dtsi b/fdts/stm32mp25-bl2.dtsi
index 0716b3eb71..784546730d 100644
--- a/fdts/stm32mp25-bl2.dtsi
+++ b/fdts/stm32mp25-bl2.dtsi
@@ -5,6 +5,10 @@
 
 / {
 	soc@0 {
+#if !STM32MP_SPI_NAND && !STM32MP_SPI_NOR
+		/delete-node/ spi@40430000;
+		/delete-node/ spi@40440000;
+#endif
 #if !STM32MP_RAW_NAND
 		/delete-node/ memory-controller@48200000;
 #endif
diff --git a/fdts/stm32mp251.dtsi b/fdts/stm32mp251.dtsi
index 882cbbff9e..a6675a6be1 100644
--- a/fdts/stm32mp251.dtsi
+++ b/fdts/stm32mp251.dtsi
@@ -99,6 +99,28 @@
 			};
 		};
 
+		ospi1: spi@40430000 {
+			compatible = "st,stm32-omi";
+			reg = <0x40430000 0x400>, <0x60000000 0x8000000>;
+			reg-names = "ospi", "ospi_mm";
+			interrupts = <GIC_SPI 163 IRQ_TYPE_LEVEL_HIGH>;
+			st,syscfg-dlyb = <&syscfg 0x1000>;
+			clocks = <&rcc CK_KER_OSPI1>;
+			resets = <&rcc OSPI1_R>, <&rcc OSPI1DLL_R>;
+			status = "disabled";
+		};
+
+		ospi2: spi@40440000 {
+			compatible = "st,stm32-omi";
+			reg = <0x40440000 0x400>, <0x68000000 0x8000000>;
+			reg-names = "ospi", "ospi_mm";
+			interrupts = <GIC_SPI 164 IRQ_TYPE_LEVEL_HIGH>;
+			st,syscfg-dlyb = <&syscfg 0x1400>;
+			clocks = <&rcc CK_KER_OSPI2>;
+			resets = <&rcc OSPI2_R>, <&rcc OSPI2DLL_R>;
+			status = "disabled";
+		};
+
 		risaf2: risaf@420b0000 {
 			compatible = "st,stm32-risaf";
 			reg = <0x420b0000 0x1000>;
-- 
2.53.0

